src/account/mod.rs

```diff
@@ -355,6 +355,9 @@ impl AccountInitialiser {
                 self.sync_accounts_lock.clone(),
             );
             drop(accounts);
+            for (_, account) in self.accounts.read().await.iter(){
+                account.accounts.write().await.insert(account_id.clone(), guard.clone());
```

Comment:
`account.accounts` should be the same reference everywhere. `account.accounts == self.accounts` i's supposed to be true. but if this is just a workaround, pls put a note above it so we can find the actual issue later.